comp/core/remoteagentregistry/impl/remoteagentregistry.go

```diff
@@ -183,14 +206,62 @@ func (ra *remoteAgentRegistry) start() {
 				}
 
 				for _, id := range agentsToRemove {
-					delete(ra.agentMap, id)
-					log.Infof("Remote agent '%s' deregistered after being idle for %s.", id, remoteAgentIdleTimeout)
+					details, ok := ra.agentMap[id]
+					if ok {
+						if details.configStream != nil && details.configStream.stream != nil {
+							details.configStream.streamCancel()
+						}
+						delete(ra.agentMap, id)
+						log.Infof("Remote agent '%s' deregistered after being idle for %s.", id, remoteAgentIdleTimeout)
+					}
 				}
 
 				ra.agentMapMu.Unlock()
 			}
 		}
 	}()
+
+	ra.conf.OnUpdate(func(setting string, oldValue, newValue any) {
+		ra.configUpdatesChan <- &settingsUpdate{
+			setting:  setting,
+			oldValue: oldValue,
+			newValue: newValue,
+		}
+	})
+	go ra.sendConfigUpdatesLoop()
```

Comment:
Any reason why adding it to a new go routine instead of adding as part of the main loop on line 193?